lib/limiter/reservation.go

```diff
@@ -0,0 +1,133 @@
+package limiter
+
+import (
+	"context"
+	"sync"
+	"tcplb/lib/core"
+)
+
+type ClientReservation struct {
+	c core.ClientID
+}
+
+// MaxReservationsExceeded is the error returned by ClientReserver
+// when an attempted reservation fails because the client has too
+// many reservations.
+var MaxReservationsExceeded error = maxReservationsExceededError{}
+
+type maxReservationsExceededError struct{}
+
+func (maxReservationsExceededError) Error() string { return "maximum client reservations exceeded" }
```

Comment:
Suggestion: simplify?

```suggestion
var MaxReservationsExceeded = errors.New("maximum client reservations exceeded")
```